From ed4398c612dc0c3672d6bee7985f5f6afc42e78e Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 2 Sep 2024 17:30:22 +0200
Subject: [PATCH 675/954] Revert "fix(psci): fix parent parsing in
 psci_is_last_cpu_to_idle_at_pwrlvl"

This reverts commit ebe8d6442c3cc1737c360e04335801ccbe31f7ed.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git c297a1cc678e61ee2ff0a1a6393683b2f661e572 Revert "fix(psci): fix parent parsing in psci_is_last_cpu_to_idle_at_pwrlvl"]
Change-Id: Iac969494ee59ec3f6ea9590c8e9b0d83727a059d
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/401562
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 lib/psci/psci_common.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/psci/psci_common.c b/lib/psci/psci_common.c
index 36e58f02e9..26b6a61e97 100644
--- a/lib/psci/psci_common.c
+++ b/lib/psci/psci_common.c
@@ -179,10 +179,9 @@ static bool psci_is_last_cpu_to_idle_at_pwrlvl(unsigned int end_pwrlvl)
 	}
 
 	my_idx = plat_my_core_pos();
-	parent_idx = my_idx;
 
 	for (lvl = PSCI_CPU_PWR_LVL; lvl <= end_pwrlvl; lvl++) {
-		parent_idx = psci_cpu_pd_nodes[parent_idx].parent_node;
+		parent_idx = psci_cpu_pd_nodes[my_idx].parent_node;
 	}
 
 	cpu_start_idx = psci_non_cpu_pd_nodes[parent_idx].cpu_start_idx;
-- 
2.53.0

